tsdfileapi/utils.py

```diff
@@ -113,3 +115,37 @@ def md5sum(filename, blocksize=65536):
         for block in iter(lambda: f.read(blocksize), b""):
             _hash.update(block)
     return _hash.hexdigest()
+
+
+def move_data_to_folder(path, dest):
+    """
+    Move file/dir at path into and folder at dest.
+
+    Parameters
+    ----------
+    path: str, uploaded file or folder
+    dest: name of the destination folder
+
+    Returns
+    -------
+    boolean
+
+    """
+    try:
+        if not dest:
+            return path
+        filename = os.path.basename(path)
+        base_dir = path.replace(f'/{filename}', '')
+        new_path = os.path.normpath(dest + '/' + filename)
+        if os.path.isdir(path):
+            if os.path.lexists(new_path):
+                # idempotency
+                shutil.rmtree(new_path)
```

Comment:
I find this somewhat scary in the case where `path` is a folder…

Reply:
It is scary, but it is correct. The use case is where someone uploads a compressed or archived directory, and the file API uncompresses/restores it on the fly. Then the new resource is a directory and not a file. And in that case, just as with the file, we have to replace it. And `cp` is not an option, especially for large resources. `mv` is just changing an inode, so it has to be like this. I'll add more tests though :)